internal/pkg/agent/operation/operator.go

```diff
@@ -141,13 +141,52 @@ func (o *Operator) Reload(rawConfig *config.Config) error {
 		return errors.New(err, "failed to unpack artifact config")
 	}
 
+	sourceURI, err := o.reloadSourceURI(rawConfig)
+	if err != nil {
+		return errors.New(err, "failed to parse source URI")
+	}
+	tmp.C.SourceURI = sourceURI
+
 	if err := o.reloadComponent(o.downloader, "downloader", tmp.C); err != nil {
 		return err
 	}
 
 	return o.reloadComponent(o.verifier, "verifier", tmp.C)
 }
 
+func (o *Operator) reloadSourceURI(rawConfig *config.Config) (string, error) {
+	type reloadConfig struct {
+		// SourceURI: source of the artifacts, e.g https://artifacts.elastic.co/downloads/
+		SourceURI string `json:"agent.download.sourceURI" config:"agent.download.sourceURI"`
+
+		// FleetSourceURI: source of the artifacts, e.g https://artifacts.elastic.co/downloads/ coming from fleet which uses
+		// different naming.
+		FleetSourceURI string `json:"agent.download.source_uri" config:"agent.download.source_uri"`
```

Comment:
Do we need to update this in config.go as well?

https://github.com/elastic/elastic-agent/blob/a0a3ed1aae5436b720b8065199afc629fe8930d6/internal/pkg/artifact/config.go#L40-L41

Handling this here instead of having this function live with the rest of the config parsing logic might mean this bug can come back in a different part of the code in the future.

Reply:
this is also an option but i was a bit reluctant as it would enable using this in our config file which does not really make sense.

Reply:
We can document that Fleet uses a different key, also we can file a follow up issue with the Fleet UI team to use the key we actually intended to use.

Reply:
I see pretty much the same code added here https://github.com/elastic/elastic-agent/pull/686/files which highly suggests we should try to handle this in a single place so we don't end up with this workaround duplicated or forgotten elsewhere in the future.

Reply:
I also have no idea where the `source_uri` key is actually coming from on the Fleet side.

Reply:
I agree with not duplicating, but I'd rather fix fleet first, then using your config here and on the reloader introduced in [#686](https://github.com/elastic/elastic-agent/pull/686).